src/pydata_sphinx_theme/toctree.py

```diff
@@ -373,40 +369,87 @@ def add_collapse_checkboxes(soup: BeautifulSoup) -> None:
+        #
+        # li.has-children
+        #   details
+        #       summary
+        #           p.caption
+        #           .toctree-toggle
+        #       ul
+        #
+        # (2) Otherwise, if the node holds a link to a page in the docs:
+        #
+        # li.has-children
+        #   a.reference
+        #   details
+        #       summary
+        #           .toctree-toggle
+        #       ul
+        #
+        # Why the difference? In the first case, the TOC section heading is not
+        # a link, but in the second case it is. So in the first case it makes
+        # sense to put the text inside the summary tag so that the user can
+        # click either the text or the .toctree-toggle chevron icon to expand/
+        # collapse the TOC subtree. But in the second case, putting the
+        # heading/link in the summary tag would make it unclear whether the
+        # clicking on the link should expand the subtree or take you to the
+        # link.
+
+        # Create <details> and put the entire subtree into it
+        details = soup.new_tag("details")
+        details.extend(element.contents)
+        element.append(details)
+
+        # Hoist the link to the top if there is one
+        toc_link = element.select_one("details > a.reference")
+        if toc_link:
+            element.insert(0, toc_link)
+
+        # Create <summary> with chevron icon
+        summary = soup.new_tag("summary")
+        span = soup.new_tag(
+            "span",
             attrs={
-                "type": "checkbox",
-                "class": ["toctree-checkbox"],
-                "id": checkbox_name,
-                "name": checkbox_name,
+                "class": "toctree-toggle",
+                "role": "presentation",  # This element and the chevron it contains are purely decorative; the actual expand/collapse functionality is delegated to the <summary> tag
             },
         )
+        span.append(soup.new_tag("i", attrs={"class": "fa-solid fa-chevron-down"}))
+        summary.append(span)
 
-        # if this has a "current" class, be expanded by default
-        # (by checking the checkbox)
-        if "current" in classes:
-            checkbox.attrs["checked"] = ""
+        # Prepend section heading (if there is one) to <summary>
+        collapsible_section_heading = element.select_one("details > p.caption")
+        if collapsible_section_heading:
+            # Put heading inside summary so that the heading text (and chevron) are both clickable
+            summary.insert(0, collapsible_section_heading)
+
+        # Prepend <summary> to <details>
+        details.insert(0, summary)
 
-        element.insert(1, checkbox)
+        # If this TOC node has a "current" class, be expanded by default
+        # (by opening the details/summary disclosure widget)
+        if "current" in classes:
+            # None is how you add a boolean HTML attribute
+            details["open"] = None
```

Comment:
Coming from the Python world, this `None` to set the boolean attribute confused me until I got to the test. Could we use the [more explicit boolean notation](https://developer.mozilla.org/en-US/docs/Glossary/Boolean/HTML) instead? 

```suggestion
            details["open"] = "open"
```